core/src/main/java/org/elasticsearch/index/mapper/internal/TTLFieldMapper.java

```diff
@@ -100,6 +101,9 @@ public TTLFieldMapper build(BuilderContext context) {
     public static class TypeParser implements MetadataFieldMapper.TypeParser {
         @Override
         public MetadataFieldMapper.Builder parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
+            if (parserContext.indexVersionCreated().onOrAfter(Version.V_5_0_0_alpha4)) {
+                throw new IllegalArgumentException("[_ttl] is removed in 5.0. As a replacement, you can delete by range query on a timestamp field.");
```

Comment:
Maybe "you should use time based indexes or cron a delete-by-query with a range query on a timestamp field"? Or something that mentions time based indexes....